Disable BusyBox bz2

stokito opened the pull request:
BB is built with enabled bzip2 which in fact not used by OpenWrt itself

See related forum thread https://forum.openwrt.org/t/is-bz2-compression-used/72120

wulfy23 commented:
it is used in image detection... which is a beneficial feature even if all targets using it become redundant... for a few Kb best to retain the capability...

possibly of use under the 'small_flash'... however for busybox... you might need a 'busybox_small' variant... which is possibly not such a silly idea... improbable for this patch alone, but stack a few up and you might be in business... 

( once you check in with the layerscape, ubnt et. al. and verify which devices are actually using bz on-device and check they are not 'small_flash' )

stokito commented:
Thank you for a quick response. What exactly you mean “used for image detection”? I made a search of “DEPENDS.*bz2” on packages and nothing special was found.
As far I understood it used only in build time.

wulfy23 commented:
```
fgrep -r bz package/base-files/ target/linux/ | grep -v '\.patch'
```

stokito commented:
Thank you for pointing out. I checked this scripts and as far I understood:
* `target/linux/layerscape/image/mkits-multiple-config.sh`  looks like used on build time
* `package/base-files/files/lib/upgrade/stage2`  directly not use but mentions `bzcat`
* `package/base-files/files/lib/upgrade/common.sh` the only one place where `bzcat` is called directly to unpack image. But from what I understood we can't compile bzipped images.

Also I found a good talk [Fight for the bytes! Fun with Four Megabytes Flash](https://youtu.be/M3yUjHKvde8?t=846) where mt mentioned that bz2 can be disabled

adschm commented:
Note that apart from the discussion about the content, this has several formal issues. Please visit https://openwrt.org/submitting-patches again...

stokito commented:
@adschm I changed commit message and signed. Please check if it's ok.

wulfy23 commented on package/utils/busybox/Config-defaults.in:
```diff
@@ -2268,6 +2268,18 @@ config BUSYBOX_DEFAULT_FEATURE_HTTPD_PROXY
 config BUSYBOX_DEFAULT_FEATURE_HTTPD_GZIP
 	bool
 	default n
+config BUSYBOX_DEFAULT_FEATURE_HTTPD_ACL_IP
+	bool
+	default n
+config BUSYBOX_DEFAULT_FEATURE_HTTPD_DATE
+	bool
+	default n
+config BUSYBOX_DEFAULT_FEATURE_HTTPD_LAST_MODIFIED
+	bool
+	default n
+config BUSYBOX_DEFAULT_FEATURE_HTTPD_ETAG
+	bool
+	default y
```
i think you have pushed something inadvertently? these do not look related to bzcar to me...

stokito replied:
Omg, yes, these changes was in another branch. I’ll fix in a hour. Sorry

fanck0605 commented:
@stokito It seems that bzip is used here [/lib/upgrade/common.sh#L99](https://github.com/openwrt/openwrt/blob/331892f85f1cfab21b26a1f84824f4110e06027f/package/base-files/files/lib/upgrade/common.sh#L99)

stokito commented:
> package/base-files/files/lib/upgrade/common.sh the only one place where bzcat is called directly to unpack image. But from what I understood we can't even compile bzipped images.

CodeFetch commented:
I also see no point in supporting bzipped images as they are not used by OpenWrt.

@stokito So maybe add a commit for removing support of bzipped images...

BTW https://github.com/openwrt/openwrt/pull/3940

stokito commented:
@CodeFetch thank you for the comment. I just not experienced in that area so I can't add a such commit. If you do feel it is needed then please send a PR for this.

CodeFetch commented:
@stokito If you want to disable bz2 support, you also need to remove the part which depends on it:
https://github.com/openwrt/openwrt/blob/331892f85f1cfab21b26a1f84824f4110e06027f/package/base-files/files/lib/upgrade/common.sh#L99

Otherwise your PR is leaving dead code and won't be merged.

stokito commented:
@CodeFetch  as you proposed I removed the support of bzipped images and force pushed the resulted commit 6ed6152
Please review

wulfy23 commented:
is this artifact intentional?
```
CONFIG_BUSYBOX_DEFAULT_BZIP2_SMALL=0
```
please provide example of claimed 8K saving

stokito committed:
busybox: disable bzip2

bzip2 adds about 8kb of size. For tiny builds it's often disabled.
It's not directly used by stock OpenWrt programs.
Kernel images compressed with bzip2 are also not fully supported.

Signed-off-by: Sergey Ponomarev <stokito@gmail.com>

stokito commented:
@wulfy23 in fact I disabled not `CONFIG_BUSYBOX_DEFAULT_BZIP2_SMALL=0` but `BUSYBOX_DEFAULT_FEATURE_BZIP2_DECOMPRESS` and yes, I did it by mistake. Thank you for noticing!

> please provide example of claimed 8kb saving
![busybox menuconfig](https://user-images.githubusercontent.com/415502/121865737-f8a51b80-cd06-11eb-8903-7d627417527b.png)
I didn't checked it myself because openwrt doesn't have `make bloatcheck` as BusyBox have.
This will also safe a build time a little bit.

aparcar commented:
Merged, thank you!

stokito commented:
Thank you all! Now we must not to forget to add notice to change log
